pkg/proxy/routes.go

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package proxy
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	"github.com/cilium/cilium/pkg/datapath/linux/route"
+)
+
+const (
+	// Mark/mask set by the bpf datapath to denote packets destined to the proxy.
+	tproxyMark = 0x200
+	tproxyMask = 0xf00
+
+	// Routing table for redirecting packets to the proxy through the local stack.
+	proxyRoutingTable = 2004
+)
+
+var (
+	// Routing rule for traffic to proxy.
+	tproxyRule = route.Rule{
+		Priority: 9,
+		Mark:     tproxyMark,
+		Mask:     tproxyMask,
+		Table:    proxyRoutingTable,
+	}
+
+	// Default IPv4 route for local delivery.
+	route4 = route.Route{
+		Table:  proxyRoutingTable,
+		Type:   route.RTN_LOCAL,
+		Local:  net.IPv4zero,
+		Device: "lo",
+		Proto:  linux_defaults.RTProto}
+
+	// Default IPv6 route for local delivery.
+	route6 = route.Route{
+		Table:  proxyRoutingTable,
+		Type:   route.RTN_LOCAL,
+		Local:  net.IPv6zero,
+		Device: "lo",
+		Proto:  linux_defaults.RTProto,
+	}
+)
+
+// installRoutesIPv4 configures routes and rules needed to redirect ingress
+// packets to the proxy.
+func installRoutesIPv4() error {
+	if err := route.Upsert(route4); err != nil {
+		return fmt.Errorf("inserting ipv4 proxy route %v: %w", route4, err)
```

Comment:
nit: change all error messages to something like `failed to insert ...`

Reply:
We usually log these as errors, and error wrapping makes 'failed' a little redundant. Anecdotally, we tend to avoid it in ebpf-go. For example, it comes out as `error="failed to insert ..."` or `x went wrong: failed to insert ...`.

IMO being terse is more readable since there's less reading to be done for unwinding the stack trace: `reticulating splines: shaving yaks: inserting ipv4 proxy route ...`, etc.